integration/testrunner/db.go

```diff
@@ -84,32 +75,34 @@ func makeBigQueryClient(ctx context.Context, tb testing.TB, projectID string) *b
 
 // This calls the database to check that an audit log exists. It uses the retries that are specified in the Config
 // file. This method assumes that only a single audit log will match, which constitutes success.
-func queryIfAuditLogExistsWithRetries(ctx context.Context, tb testing.TB, bqQuery *bigquery.Query, cfg *Config, testName string) {
+func queryIfAuditLogExistsWithRetries(ctx context.Context, tb testing.TB, bqQuery *bigquery.Query, cfg *Config, testName string) []bigquery.Value {
 	tb.Helper()
 
-	queryIfAuditLogsExistWithRetries(ctx, tb, bqQuery, cfg, testName, 1)
+	return queryIfAuditLogsExistWithRetries(ctx, tb, bqQuery, cfg, testName)
 }
 
 // This calls the database to check that an audit log exists. It uses the retries that are specified in the Config
 // file. This method allows for specifying how many logs we expect to match, in order to handle streaming use cases.
-func queryIfAuditLogsExistWithRetries(ctx context.Context, tb testing.TB, bqQuery *bigquery.Query, cfg *Config, testName string, expectedNum int64) {
+func queryIfAuditLogsExistWithRetries(ctx context.Context, tb testing.TB, bqQuery *bigquery.Query, cfg *Config, testName string) []bigquery.Value {
 	tb.Helper()
-
+	var result []bigquery.Value
 	b := retry.NewExponential(cfg.LogRoutingWait)
 	if err := retry.Do(ctx, retry.WithMaxRetries(cfg.MaxDBQueryTries, b), func(ctx context.Context) error {
-		found, err := queryIfAuditLogExists(ctx, tb, bqQuery, expectedNum)
-		if found {
+		row, err := queryIfAuditLogExists(ctx, tb, bqQuery)
+		if row != nil {
 			// Early exit retry if queried log already found.
+			result = row
 			return nil
 		}
 
 		tb.Log("Matching entry not found, retrying...")
 
 		if err != nil {
-			tb.Logf("Query error: %v.", err)
+			tb.Logf("query error: %s", err.Error())
```

Comment:
Should we return error here? As we discussed in debug sessions, this _usually_ means that the query itself has problem. And we only want to retry if the expected logs were not found.